autometa/common/metabin.py

```diff
@@ -127,70 +285,73 @@ def get_orfs(self, orf_type='prot'):
 
         Raises
         -------
+        KeyError
+            `orf_type` not in ['prot','nucl']
         FileNotFoundError
-            Why the exception is raised.
+            Either `self.prot_orfs_fpath` or `self.nucl_orfs_fpath` not found.
+            This will correspond to `orf_type`
 
         """
         orfs_fpaths = {'prot':self.prot_orfs_fpath, 'nucl':self.nucl_orfs_fpath}
-        orfs_fpath = orfs_fpaths.get(orf_type, 'prot')
+        if orf_type not in orfs_fpaths:
+            raise KeyError(f'{orf_type} not in {orfs_fpaths}')
+        orfs_fpath = orfs_fpaths[orf_type]
         if not self.prepared(orfs_fpath):
             raise FileNotFoundError(orfs_fpath)
         return prodigal.orf_records_from_contigs(
             contigs=self.contigs,
             fpath=self.prot_orfs_fpath)
 
     def write_orfs(self, fpath, orf_type='prot'):
+        """Write `orf_type` ORFs from `contigs` to `fpath`.
+
+        Parameters
+        ----------
+        fpath : str
+            </path/to/write/`orf_type`.orfs.fasta
+        orf_type : str
+            type of ORFs to write, choices=['prot','nucl']
+
+        """
         records = self.get_orfs(orf_type=orf_type)
         SeqIO.write(records, fpath, 'fasta')
 
-    def get_seqs(self, all=False):
-        """Retrieve sequences from assembly.
+    def describe(self, autometa_details=True):
+        """Describe various statistics of the MetaBin instance.
 
         Parameters
         ----------
-        all : bool
-            Gets all sequences from assembly if True else sequences for MAG
-            (the default is False).
-
-        Returns
-        -------
-        list
-            list of SeqIO [SeqRecords, ...]
+        autometa_details : bool
+            If True, will output autometa related information (the default is True).
 
         """
-        if all:
-            return [seq for seq in SeqIO.parse(self.assembly, 'fasta')]
-        return [seq for seq in SeqIO.parse(self.assembly, 'fasta')
-            if seq.id in self.contigs]
 
-    def describe(self, autometa_details=True):
-        print(f'''
-M.A.G. Details
-________________________
-Num. Contigs: {self.nseqs:,} / {self.nallseqs:,} ({self.seqs_pct:4.2f}% of total metagenome)
-Num. Nucl. ORFs: {self.nnucls:,}
-Num. Prot. ORFs: {self.nprots:,}
-Size: {self.size:,}bp / {self.totalsize:,}bp ({self.size_pct:4.2f}% of total metagenome)
-Mean GC content: {self.gc_content:4.2f}%
-________________________
-''')
+        print('Metabin Details\n'
+            '________________________\n'
+            f'Num. Contigs: {self.nseqs:,} / {self.nallseqs:,} ({self.seqs_pct:4.2f}% of total metagenome)\n'
+            f'Num. Nucl. ORFs: {self.nnucls:,}\n'
+            f'Num. Prot. ORFs: {self.nprots:,}\n'
+            f'Size: {self.size:,}bp / {self.totalsize:,}bp ({self.size_pct:4.2f}% of total metagenome)\n'
+            f'Mean GC content: {self.length_weighted_gc:4.2f}%\n'
```

Comment:
The print output should probably explicitly say the GC content is length-weighted.